src/LiveComponent/src/Hydration/ArrayCollectionHydrationExtension.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Hydration;
+
+use Doctrine\Common\Collections\ArrayCollection;
+
+/**
+ * @author Jean-Paul van der Wegen <info@jpvdw.nl>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+class ArrayCollectionHydrationExtension extends AbstractDoctrineHydrationExtension implements HydrationExtensionInterface
+{
+    public function supports(string $className): bool
+    {
+        return ArrayCollection::class === $className;
```

Comment:
So this class supports every ArrayCollection, whatever the entity inside ?

Reply:
It should be persistent entities. I have converted abstract class into a trait. Within the trait, the function findObject has been changed to getObject, and if it is not a persistent entity, exception "EntityNotFoundException" will be given

For non-persistent entities, I believe the ArrayCollection is not the right solution. The LiveCollectionType should be used for this usecase ?